static/js/stream_edit.js

```diff
@@ -306,10 +306,26 @@ function show_subscription_settings(sub) {
             sub.stream_id +
             "'] .pill-container",
     );
+
+    function create_item_from_text(text, current_items) {
+        const item = stream_pill.create_item_from_stream_name(text, current_items);
+        if (item) {
+            return item;
+        }
+        return user_pill.create_item_from_email(text, current_items);
+    }
+    function get_text_from_item(item) {
+        const text = stream_pill.get_stream_id_from_item(item);
+        if (text) {
+            return text;
+        }
+        return user_pill.get_email_from_item(item);
+    }
```

Comment:
I don't really understand what this commit does; is this interface for copy-pasting content into the box?

Reply:
Yes besides copy pasting, this last commit is added so that stream pills can be created when the user does not hides the typeahead (presses `Esc` key before enter). The user pill also has this feature.

( I updated the commit )